### PR TITLE
[MXNET-647] Add Diff Collator

### PR DESCRIPTION
**Pull request by cetsai**
## Description ##
This is a script that returns a list of changes between the current branch and the master branch. This is going to be a part of the upcoming automated flaky test detection system, but it can also be used as a standalone tool. 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Added Diff Collator

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by cetsai** [comment]
@haojin2 Hey, let me know if this looks good when you have a moment

File: tools/diff_collator.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    args = arg_parser.parse_args()
+    return args
+
+
+if __name__ == "__main__":
+    
+    args = parse_args()
+
+    try:
+        if args.commits is  not None:
```
Comment (haojin2) [inline tools/diff_collator.py]: nit: An unnecessary extra space here?

File: tools/diff_collator.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Output a list of differences between current git branch and master
+
+Precondition: this script is run instide an existing git repository
```
Comment (haojin2) [inline tools/diff_collator.py]: typo: instide

File: tools/diff_collator.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+This script will performs a retrieves and output a list of changes
+based on the output of git diff. For each changes, the file, line numbers
+and top-level funtcion name is provided.
```
Comment (haojin2) [inline tools/diff_collator.py]: typo: funtcion
Plus this is a bit hard to understand, please re-phrase these a bit.

**Comment by cetsai** [comment]
These changes will be submitted in another pull request once the automated flaky test detector is finished.

**Pull request closed without merging**